aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/XrayRulesSampler.java

```diff
@@ -70,4 +96,46 @@ public SamplingResult shouldSample(
   public String getDescription() {
     return "XrayRulesSampler{" + Arrays.toString(ruleAppliers) + "}";
   }
+
+  List<GetSamplingTargetsRequest.SamplingStatisticsDocument> snapshot(Date now) {
+    return Arrays.stream(ruleAppliers)
+        .map(rule -> rule.snapshot(now))
+        .filter(Objects::nonNull)
+        .collect(Collectors.toList());
+  }
+
+  long nextTargetFetchTimeNanos() {
+    return Arrays.stream(ruleAppliers)
+        .mapToLong(SamplingRuleApplier::getNextSnapshotTimeNanos)
+        .min()
```

Comment:
Does this do away with the constant-interval polling?

Reply:
Yeah - given we have intervals for the targets in the responses themselves there doesn't seem to be a need to fixed-poll targets